d3a_interface/sim_results/kpi.py

```diff
@@ -145,11 +146,59 @@ def update_area_kpi(self, area_dict, core_stats):
         self._accumulate_energy_trace(core_stats)
 
 
+class SavingKPI:
+    def __init__(self):
+        self.producer_list = list()
+        self.consumer_list = list()
+        self.ess_list = list()
+        self.house_list = list()
+        self.fit_revenue = 0.
+        self.utility_bill = 0.
+        self.d3a_revenue = 0.
+        self.saving_absolute = 0.
+        self.saving_percentage = 0.
+
+    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
+        """
+        :param area_dict: contain nested area info
+        :param core_stats: contain area's raw/key statistics
+        :param gf_alp: grid_fee_along_the_path - cumulative grid fee from root to target area
+        :return:
```

Comment:
As discussed on slack, please use Google-style docstrings rather than reStructured Text.

You can change the PyCharm settings from:
`Settings > Python integrated tools > Docstrings > Docstring format`